### PR TITLE
🌱 attestor: e2e tests

### PR DESCRIPTION
**Pull request by raghavkaul**
#### What kind of change does this PR introduce?
Add E2E tests. Also refactor the check runner to allow dependency injection.

### CODE REVIEW

File: e2e/attestor_policy_test.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+					policy: policy.AttestationPolicy{
+						PreventBinaryArtifacts:      true,
+						AllowedBinaryArtifacts:      []string{"test-binary-artifact-*"},
+						PreventKnownVulnerabilities: true,
+						PreventUnpinnedDependencies: false,
+					},
+					expected: policy.Pass,
+				},
+				{
+					name:    "test bad repo with ignored dep by path",
```
Comment (azeemshaikh38) [inline e2e/attestor_policy_test.go:71]: Add failing test for unpinned dep?

File: e2e/attestor_policy_test.go [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+						EnsureCodeReviewed: true,
+						CodeReviewRequirements: policy.CodeReviewRequirements{
+							MinReviewers: 1,
+						},
+					},
+					expected: policy.Pass,
+				},
+				{
+					name:    "test code reviews required but repo doesn't have code reviews",
+					repoURL: "https://github.com/ossf-tests/scorecard-binauthz-test-bad",
```
Comment (azeemshaikh38) [inline e2e/attestor_policy_test.go:115]: Few more tests I can think of for code review: (1) when only `required_reviewers` is given, (2) when both `min_reviewers` and `required_reviewers` is given.

File: attestor/command/check.go [hunk 3]
```diff
@@ -35,6 +35,11 @@ func (ep EmptyParameterError) Error() string {
 }
 
 func runCheck() (policy.PolicyResult, error) {
+	return RunCheckWithParams(repoURL, commitSHA, policyPath)
+}
+
+// RunCheckWithParams: Run scorecard check on repo. Export for testability.
+func RunCheckWithParams(repoURL, commitSHA, policyPath string) (policy.PolicyResult, error) {
```
[the remark this answers is not in the document]
Comment (raghavkaul) [reply attestor/command/check.go:42]: This needs to be exported, since the e2e tests are in a different package

**Review by olivekl — COMMENTED** [review]
Doc change LGTM.

**Review by azeemshaikh38 — APPROVED** [review]
Thanks!

**Pull request merged**